src/screens/Analytics/AnalyticsNew.res

```diff
@@ -0,0 +1,714 @@
+        dict->Dict.get(key)
+      })
+      ->Array.joinWithUnsafe("")
+    }
+
+    open AnalyticsTypes
+    let getUpdatedData = (data, weeklyData, cols) => {
+      let dataArr = data->parseData
+      let weeklyArr = weeklyData->parseData
+
+      dataArr
+      ->Array.map(item => {
+        let dataDict = item->getDictFromJsonObject
+        let dataKey = activeTab->generateIDFromKeys(dataDict)
+
+        weeklyArr->Array.forEach(newItem => {
+          let weekklyDataDict = newItem->getDictFromJsonObject
+          let weekklyDataKey = activeTab->generateIDFromKeys(weekklyDataDict)
+
+          if dataKey === weekklyDataKey {
+            cols->Array.forEach(
+              obj => {
+                switch weekklyDataDict->Dict.get(obj.refKey) {
+                | Some(val) => dataDict->Dict.set(obj.newKey, val)
+                | _ => ()
+                }
+              },
+            )
+          }
+        })
+        dataDict->JSON.Encode.object
+      })
+      ->JSON.Encode.array
+      ->getTable
+      ->Array.map(Nullable.make)
+    }
+
+    open Promise
+    let getWeeklyData = async (data, cols) => {
+      let weeklyDateRange = HSwitchRemoteFilter.getDateFilteredObject()
+
+      let weeklyTableReqBody = AnalyticsUtils.generateTablePayload(
+        ~startTimeFromUrl=weeklyDateRange.start_time,
+        ~endTimeFromUrl=weeklyDateRange.end_time,
+        ~filterValueFromUrl,
+        ~currenltySelectedTab=activeTab,
+        ~deltaMetrics,
+        ~isIndustry=false,
+        ~distributionArray=None,
+        ~deltaPrefixArr=deltaArray,
+        ~tableMetrics=[],
+        ~mode=None,
+        ~customFilter,
+        ~moduleName,
+        ~showDeltaMetrics=true,
+        (),
+      )
+
+      fetchDetails(tableEntity.uri, weeklyTableReqBody, Post, ())
+      ->thenResolve(json => {
+        setTableData(_ => getUpdatedData(data, json, cols))
+        setTableDataLoading(_ => false)
+        setShowTable(_ => true)
+      })
+      ->catch(_ => {
+        setTableDataLoading(_ => false)
+        resolve()
+      })
+      ->ignore
+    }
+
+    React.useEffect3(() => {
+      setShowTable(_ => false)
+      if (
+        startTimeFromUrl->LogicUtils.isNonEmptyString && endTimeFromUrl->LogicUtils.isNonEmptyString
+      ) {
+        let tableReqBody = HSAnalyticsUtils.generateTablePayload(
+          ~startTimeFromUrl,
+          ~endTimeFromUrl,
+          ~filterValueFromUrl,
+          ~currenltySelectedTab=activeTab,
+          ~deltaMetrics,
+          ~isIndustry=false,
+          ~distributionArray,
+          ~deltaPrefixArr=deltaArray,
+          ~tableMetrics=[],
+          ~mode=None,
+          ~customFilter,
+          ~moduleName,
+          ~showDeltaMetrics=true,
+          (),
+        )
+
+        fetchDetails(tableEntity.uri, tableReqBody, Post, ())
+        ->thenResolve(json => {
+          switch weeklyTableMetricsCols {
+          | Some(cols) => getWeeklyData(json, cols)->ignore
```

Comment:
please recheck this function. try moving this out of useEffect and call as separate function.code is bit not readable